graylog-storage-opensearch2/src/main/java/org/graylog/storage/opensearch2/PaginationOS2.java

```diff
@@ -46,8 +48,13 @@ public PaginationOS2(final ResultMessageFactory resultMessageFactory,
     public ChunkedResult retrieveChunkedResult(final ChunkCommand chunkCommand) {
         final SearchSourceBuilder searchQuery = searchRequestFactory.create(chunkCommand);
         final SearchRequest request = buildSearchRequest(searchQuery, chunkCommand.indices());
-        final SearchResponse result = client.search(request, "Unable to perform search-after pagination search");
-        return new PaginationResultOS2(resultMessageFactory, client, request, result, searchQuery.toString(), chunkCommand.fields(), chunkCommand.limit().orElse(-1));
+        // doing a msearch so that it results in a POST so we don't have to deal with possible errors where the request exceeds HTTP parameter lengths
+        final var result = client.msearch(List.of(request), "Unable to perform search-after pagination search");
```

Comment:
Sorry if I am asking a stupid question, but it seams that `client.search` calls multisearch anyway: https://github.com/Graylog2/graylog2-server/blob/master/graylog-storage-opensearch2/src/main/java/org/graylog/storage/opensearch2/OpenSearchClient.java#L89

So why does it help?